main.tf

```diff
@@ -47,6 +47,7 @@ locals {
   namespace_list = [
     for p in var.drupal_projects_list : {
       namespace = p.kubernetes_namespace == null ? "${p.project_name}-${p.gitlab_project_id}-${p.release_branch_name}" : p.kubernetes_namespace
+      labels    = p.kubernetes_namespace_labels == null ? {} : p.kubernetes_namespace_labels
```

Comment:
I think you can avoid the check on `p.kubernetes_namespace_labels == null` because the default is `{}`.